src/test/java/educationalproject/programmingstuff/tests_integrations/EndpointsProcessingTests.java

```diff
@@ -0,0 +1,171 @@
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class EndpointsProcessingTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ItemRepository itemRepository;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private UserResponseMapper userResponseMapper;
+
+    @Autowired
+    private UserService userService;
+
+    private RequestBuilder request;
+
+    private List<User> userEntities;
+
+    private List<UserResponseDto> userResponseDtos;
+
+    private String expectedResponse;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeEach
+    private void prepareTestData() {
+
+        List<User> users= TestDataCreator.getTestUsers();
+        userRepository.saveAllAndFlush(users);
+
+    }
+
+    @Test
+    @Transactional
+    void givenGetEndpointUsersWithNoParam_whenEndpointTriggered_thenReturnAllUsers() throws Exception {
+
+        //Given
+        request = MockMvcRequestBuilders.get("/users");
+        userEntities = userRepository.findAll();
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenGetEndpointUsersWithName_whenEndpointTriggered_thenReturnUsersWithThisName() throws Exception {
+
+        //Given
+        String testName = "John";
+        request = MockMvcRequestBuilders.get("/users").param("name", testName);
+        userEntities = userRepository.getUsersByName(testName);
+        userResponseDtos = userResponseMapper.makeUsersResponseOf(userEntities);
+        expectedResponse = objectMapper.writeValueAsString(userResponseDtos);
+
+        //When
+        ResultActions resultActions = mockMvc.perform(request);
+
+        //Then
+        resultActions
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedResponse));
+    }
+
+    @Test
+    @Transactional
+    void givenPostEndpointUsersWithNewUser_whenEndpointTriggered_thenReturnThatUserWithNewId() throws Exception {
+
+        //Given
+
+        String testName = "Sasha";
+        String testSurname = "Grey";
```

Comment:
okay, boomer:)

Reply:
So you recognize it. You are boomer as well. :)